dart/lib/src/hub.dart

```diff
@@ -0,0 +1,379 @@
+import 'dart:async';
+import 'dart:collection';
+
+import 'client.dart';
+import 'protocol.dart';
+import 'scope.dart';
+import 'sentry_options.dart';
+
+typedef ScopeCallback = void Function(Scope);
+
+/// SDK API contract which combines a client and scope management
+class Hub implements HubInterface {
+  static SentryClient _getClient({SentryOptions fromOptions}) {
+    return SentryClient(
+      dsn: fromOptions.dsn,
+      environmentAttributes: fromOptions.environmentAttributes,
+      compressPayload: fromOptions.compressPayload,
+      httpClient: fromOptions.httpClient,
+      clock: fromOptions.clock,
+      uuidGenerator: fromOptions.uuidGenerator,
+    );
+  }
+
+  final ListQueue<_StackItem> _stack;
+
+  final SentryOptions _options;
+
+  factory Hub(SentryOptions options) {
+    _validateOptions(options);
+
+    return Hub._(options);
+  }
+
+  Hub._(SentryOptions options)
+      : _options = options,
+        _stack = ListQueue() {
+    _stack.add(_StackItem(_getClient(fromOptions: options), Scope(_options)));
+    _isEnabled = true;
+  }
+
+  static void _validateOptions(SentryOptions options) {
+    if (options == null) {
+      throw ArgumentError.notNull('options');
+    }
+
+    if (options.dsn == null) {
+      throw ArgumentError.notNull('options.dsn');
+    }
+  }
+
+  bool _isEnabled = false;
+
+  @override
+  bool get isEnabled => _isEnabled;
+
+  SentryId _lastEventId;
+
+  @override
+  SentryId get lastEventId => _lastEventId;
+
+  @override
+  Future<SentryId> captureEvent(Event event) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureEvent' call is a no-op.",
+      );
+    } else if (event == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureEvent called with null parameter.',
+      );
+    } else {
+      final item = _stack.last;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureEvent(event: event);
+        } catch (err) {
+          /* FIXME(rxlabz) ? Event.id ?*/
```

Comment:
I suggest avoiding FIXME but instead use TODO, it's easier to find out (IDE helps), also to not use (rxlabz) because I could help on it too